GovUk.Frontend.Umbraco/Blocks/BlockViewService.cs

```diff
@@ -0,0 +1,183 @@
+                {
+                    nextRowClass = _gridClassBuilder.BuildGridRowClasses(blocks[i + 1].Settings?.Value<string>(PropertyAliases.CssClassesForRow));
+                    nextColumnClass = _gridClassBuilder.BuildGridColumnClasses(
+                                                            blocks[i + 1].Settings?.Value<string>(PropertyAliases.ColumnSize),
+                                                            blocks[i + 1].Settings?.Value<string>(PropertyAliases.ColumnSizeFromDesktop),
+                                                            blocks[i + 1].Settings?.Value<string>(PropertyAliases.CssClassesForColumn),
+                                                            blocks[i + 1].Content.ContentType.Alias,
+                                                            childColumnsDefaultToFullWidth);
+
+                    var nextIsGridAreasBlock = (blocks[i + 1].Areas.Any());
+                    sameAsNext = (hasGridAreas == nextIsGridAreasBlock &&
+                                  rowClass == HtmlClassNames.Row && nextRowClass == HtmlClassNames.Row &&
+                                  columnClass == nextColumnClass);
+                }
+
+                blocksToReturn.Add(new BlockViewModel
+                {
+                    Block = blocks[i],
+                    ColumnClasses = columnClass,
+                    RowClasses = rowClass,
+                    HasGridAreas = hasGridAreas,
+                    IsGridRow = false,
+                    RenderGrid = false,
+                    IsSameAsNext = sameAsNext,
+                    IsSameAsPrevious = sameAsPrevious,
+                    FieldsetErrorClasses = FieldsetErrorClassesForBlock(_fieldsetErrorFinder, modelState, blocks[i])
+                });
+
+                previousIsGridAreasBlock = hasGridAreas;
+                previousRowClass = rowClass;
+                previousColumnClass = columnClass;
+            }
+
+            return blocksToReturn;
+        }
+
+        private static string? FieldsetErrorClassesForBlock(IGovUkFieldsetErrorFinder _fieldsetErrorFinder, ModelStateDictionary modelState, IOverridableBlockReference<IOverridablePublishedElement, IOverridablePublishedElement> block)
+        {
+            // If this block is a fieldset and there is a fieldset-level error, add extra classes to show that the entire fieldset is in an error state.
+            // But only if the 'legendIsPageHeading' setting is false, otherwise it's done in GovUkFieldset.cshtml.
+            var fieldsetErrors = _fieldsetErrorFinder.FindErrors(block, modelState);
+            string? fieldsetErrorClasses = null;
+            if (fieldsetErrors.Any())
+            {
+                var legendIsPageHeading = block.Settings?.Value<bool>(PropertyAliases.FieldsetLegendIsPageHeading) ?? false;
+                if (!legendIsPageHeading)
+                {
+                    fieldsetErrorClasses = "govuk-form-group govuk-form-group--error";
```

Comment:
Text used in multiple files. Maybe worth using a constant here?

Reply:
Done